lib/api/decorators/single.rb

```diff
@@ -51,6 +51,29 @@ def initialize(model, context = {})
                exec_context: :decorator,
                render_nil: false
 
+      def self.self_link(path, title_getter: -> (*) { represented.name })
+        link :self do
+          link_object = { href: api_v3_paths.send(path, represented.id) }
+          link_object[:title] = instance_eval(&title_getter)
+
+          link_object
+        end
+      end
+
+      def self.linked_property(property,
+                               path: property,
+                               backing_field: property,
```

Comment:
In Rails terms, this would be an association. From what I have seen, I think that the method is limited to representing associations, so renaming it to such would make sense. 
By the way, if backing_field is not set, you could also take property as a default. Same is true for path.